runtime/j9vm/jvm.c

```diff
@@ -1982,6 +1981,7 @@ JNI_CreateJavaVM_impl(JavaVM **pvm, void **penv, void *vm_args, BOOLEAN isJITSer
 	 * See the discussion in https://github.com/eclipse-openj9/openj9/pull/14634.
 	 */
 	specialArgs.captureCommandLine = FALSE;
+	iconv_init();
```

Comment:
I think this makes the call in `preloadLibraries()` redundant: the call there should be removed.

There's also a call in the redirector, so even this is normally redundant (unless a JNI application bypasses the redirector), but until the main shared library is renamed (see #7355), this should stay.

Reply:
> I think this makes the call in preloadLibraries() redundant: the call there should be removed.

I see `preloadLibraries()` is called from places like get `GetStringPlatform()`, `NewStringPlatform()`, not sure if they will always go throw the code here.

> There's also a call in the redirector, so even this is normally redundant (unless a JNI application bypasses the redirector)

I added the call because I see `JNI_CreateJavaVM_impl()` is called directly by `JITServer_CreateServer()`, I guess that does not call though the redirector.

Reply:
I don't think it's legal to call `GetStringPlatform()` or `NewStringPlatform()` before `JNI_CreateJavaVM()` so this new call to  `iconv_init()` should be sufficient, but extra calls to `iconv_init()` are sufficiently inexpensive that it probably isn't worth worrying about. If you want to keep the call in `preloadLibraries()` then this one should only be for Java 21+.

Reply:
> If you want to keep the call in preloadLibraries() then this one should only be for Java 21+.

Added this call only for Java 21+.